### PR TITLE
Data model for lcov

### PR DESCRIPTION
**Pull request by vagisha-nidhi**
Creating Lcov file format from Code coverage reader data model.
Added test for the same.

### CODE REVIEW

File: LcovGenerator/LcovModel.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        public Dictionary<string, Dictionary<int, int>> Data;
+
+        public LcovModel()
+        {
+
+        }
+
+        public LcovModel(Dictionary<string, Dictionary<int, int>> lcovModelData)
+        {
+          this.Data = lcovModelData;
```
Comment (ShreyasRmsft) [inline LcovGenerator/LcovModel.cs]: nit: indentation

File: LcovGenerator/LcovModel.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    public class Generator
+    {
+        public LcovModel ModelConverter(CodeCoverageDataModel codeCoverageDataModel)
+        {
+            Dictionary<string, Dictionary<int, int>> lcovModelData = new Dictionary<string, Dictionary<int, int>>();
+
+            foreach (var range in codeCoverageDataModel.Ranges)
+            {
+                var source = codeCoverageDataModel.Sources[range.SourceFileId];
+
```
Comment (ShreyasRmsft) [inline LcovGenerator/LcovModel.cs]: nit: extra newline

File: LcovGeneratorTests/LcovGeneratorTests.csproj [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.7.0" />
+    <PackageReference Include="MSTest.TestAdapter" Version="1.2.1" />
```
Comment (ShreyasRmsft) [inline LcovGeneratorTests/LcovGeneratorTests.csproj]: bump version to 1.3.2

Comment (ShreyasRmsft) [reply LcovGeneratorTests/LcovGeneratorTests.csproj]: and net sdk to 15.7.8

Comment (ShreyasRmsft) [reply LcovGeneratorTests/LcovGeneratorTests.csproj]: for all projects

File: src/LcovGeneratorTests/LcovGeneratorTests.csproj [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.7.0" />
```
Comment (ShreyasRmsft) [inline src/LcovGeneratorTests/LcovGeneratorTests.csproj]: can we bump these up to the latest please

File: src/LcovGeneratorTests/LcovGeneratorTests.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            Dictionary<string, Dictionary<int, int>> expecedLcovModelData = new Dictionary<string, Dictionary<int, int>>();
+            expecedLcovModelData.Add("s1.cs", new Dictionary<int, int> { { 5, 2 }, { 6, 2 } });
+            expecedLcovModelData.Add("s2.cs", new Dictionary<int, int> { { 9, 1 }, { 10, 1 }, { 11, 1 }, { 12, 1 } });
+
+            var actualLcovModelData = generator.ModelConverter(codeCoverageDataModel);
+
+            var actualText = actualLcovModelData.ToString();
+            System.Console.WriteLine(actualText);
+
+            string expectedText = "SF:s1.cs" + Environment.NewLine +
```
Comment (smadala) [inline src/LcovGeneratorTests/LcovGeneratorTests.cs]: Move the expected text to file. To make it readable. Tests should help us to doc.

File: src/LcovGeneratorTests/LcovGeneratorTests.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                "DA:5,2" + Environment.NewLine +
```
Comment (smadala) [inline src/LcovGeneratorTests/LcovGeneratorTests.cs]: What is `DA`?

File: src/LcovGeneratorTests/LcovGeneratorTests.cs [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+                "DA:6,2" + Environment.NewLine +
+                "end_of_record" + Environment.NewLine +
+                "SF:s2.cs" + Environment.NewLine +
+                "DA:9,1" + Environment.NewLine +
+                "DA:10,1" + Environment.NewLine +
+                "DA:11,1" + Environment.NewLine +
```
Comment (smadala) [inline src/LcovGeneratorTests/LcovGeneratorTests.cs]: Add not covered scenario.

File: src/LcovGeneratorTests/LcovGeneratorTests.csproj [hunk 8]
```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
```
Comment (smadala) [inline src/LcovGeneratorTests/LcovGeneratorTests.csproj]: Add
```xml
 <PropertyGroup>
    <VstestCoverageRoot Condition="$(VstestCoverageRoot) == ''">..\..\</VstestCoverageRoot>
    <TestProject>true</TestProject>
  </PropertyGroup>
```
Remove
```xml
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.8.0" />
    <PackageReference Include="MSTest.TestAdapter" Version="1.3.2" />
    <PackageReference Include="MSTest.TestFramework" Version="1.3.2" />
```

File: src/LcovGenerator/LcovModel.cs [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    public class LcovModel
+    {
+        public Dictionary<string, Dictionary<int, int>> Data;
+
+        public LcovModel()
+        {
+
+        }
+
+        public LcovModel(Dictionary<string, Dictionary<int, int>> lcovModelData)
```
Comment (smadala) [inline src/LcovGenerator/LcovModel.cs]: Add documentation.

File: src/LcovGeneratorTests/LcovGeneratorTests.csproj (hunk 8 above)
Comment (ShreyasRmsft) [reply src/LcovGeneratorTests/LcovGeneratorTests.csproj]: also needed to add  <Import Project="$(VstestCoverageRoot)scripts/build/Vstest.Coverage.Settings.targets" />

**Comment by codito** [comment]
Can we close this? Dupe of https://github.com/spekt/vstest.coverage/pull/24?

**Pull request closed without merging**